### DISCUSSION TITLE
Message being delayed and the incoming message and outgoing message rate chart show info that's not expected

### DISCUSSION THREAD

**Discussion by ducanh2110** [emqx/emqx#11647] [Q&A]
Hi emqx supporters, dev, community.
I am using EMQX v5.1.6. Mine has 3 node, 3 replicant. Each VM has 8CPU, 8 GB ram, XEON 6342 Gold 2.8GHZ.
My configuration for testing right now is: I make 27000 ccu listening on a topic through an LB route to 3 replicant. 1 application will publish to EMQX with the incoming rate showing in the dashboard is: 70-100message/second. I published at QOS(0). I 've tried all the tuning guide in perfomance tuning in docs. The topic subscriber listen will receive 50-70message. But I can't reach the outgoing rate as my expectation. It's only 1.1 millions message/s on 3 replicant. My network bandwidth is 40Gbps. I've calculate in each replicant the Tx/Rx the peek size: 700-800MB.
![image_1695276654](https://github.com/emqx/emqx/assets/13704782/d0639680-a8e3-4e08-a252-3a2ad3c3f800)

One more problem is:
After running for 30 min. My application will reduce the message publish to 0-5 messages. But the outgoing at this time still be 1.1 millions and not stoping for 10-15 minute.And my subscriber at this time has received so many old messages (some messages be delivered after 30 minutes from publishing)

What did you expect to happen?
I expected that the outgoing rate will be 1350000 - 1890000 message/s. And all the message will be realtime. What can I do to raise the outgoing rate and make it not delays like that. Is there any configuration that's can be config?

How can we reproduce it (as minimally and precisely as possible)?
The size of each message i am publishing is 70byte - 120byte and being encoded base64. You guys can reproduce like that will my test case above.

EMQX version
Details
sysdescr  : EMQX
version   : 5.1.6
datetime  : 2023-09-21T13:54:16.886454592+07:00
uptime    : 21 days, 23 hours, 35 minutes, 33 seconds
OS version
Details
# On Linux:
$ cat /etc/os-release
NAME="Red Hat Enterprise Linux"
VERSION="8.8 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.8 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.8
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.8"
$ uname -a
Linux fdev-price-streaming-new02 4.18.0-477.15.1.el8_8.x86_64 #1 SMP Fri Jun 2 08:27:19 EDT 2023 x86_64 x86_64 x86_64 GNU/Linux

THANKS SO MUCH

**Comment by SergeTupchiy** [comment]
Replied in https://github.com/emqx/emqx/issues/11646
